lib/sidekiq/options.rb

```diff
@@ -3,16 +3,23 @@
 module Sidekiq
   module Options
     def self.[](key)
-      new_version? ? Sidekiq[key] : Sidekiq.options[key]
+      options_field ? Sidekiq.public_send(options_field)[key] : Sidekiq[key]
     end
 
     def self.[]=(key, value)
-      new_version? ? Sidekiq[key] = value : Sidekiq.options[key] = value
+      options_field ? Sidekiq.public_send(options_field)[key] = value : Sidekiq[key] = value
     end
 
-    # sidekiq --version >= 6.5.0
-    def self.new_version?
-      @new_version ||= Sidekiq.respond_to?(:[])
+    def self.options_field
+      return @options_field unless @options_field.nil?
+      sidekiq_version = Gem::Version.new(Sidekiq::VERSION)
+      @options_field = if sidekiq_version >= Gem::Version.new('7.beta')
```

Comment:
`Gem::Version` being weird about versions with beta/rc may make it preferable to wait until `7.0` is officially released to merge. Wasn't able to figure out a clean way to do this check